gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/Oplog.java

```diff
@@ -7865,8 +7995,8 @@ public void initialize(byte opCode,
 
       if (needsKey) {
         Object key = entry.getKeyCopy();
-        this.keyBytes = EntryEventImpl.serialize(key);
-        this.size += (4 + this.keyBytes.length);
+        this.keyBytes = EntryEventImpl.serializeDirect(key, null);
```

Comment:
Does it help to keep the key on DirectByteBuffer? I remember discussint that small objects can cause fragmentations.